File: BaeYoungSuk/Q32.java
import java.util.*;

class Solution {
    public int[] solution(String[] operations) {
        PriorityQueue<Integer> minpq = new PriorityQueue<>((o1, o2) -> o1 - o2);
        PriorityQueue<Integer> maxpq = new PriorityQueue<>((o1, o2) -> o2 - o1);
        
        for(String e : operations){
            StringTokenizer st = new StringTokenizer(e);
            String command = st.nextToken();
            int num = Integer.parseInt(st.nextToken());
            if(command.equals("I")){
                minpq.offer(num);
                maxpq.offer(num);
            } else if (command.equals("D")){
                if(num == 1 && !maxpq.isEmpty()){
                    int target = maxpq.poll();
                    minpq.remove(target);
                } else if(num == -1 && !minpq.isEmpty()){
                    int target = minpq.poll();
                    maxpq.remove(target);
                }
            }
        }
        
        int[] answer = new int[2];
        if(!maxpq.isEmpty()){
            answer[0] = maxpq.poll();
            answer[1] = minpq.poll();
        } else {
            answer[0] = 0;
            answer[1] = 0;
        }
        
        return answer;
    }
}

File: rt3310/Q32.cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(vector<string> operations) {
    vector<int> answer;
    priority_queue<int> maxpq;
    priority_queue<int> minpq;

    for (string op : operations) {
        if (op[0] == 'I') {
            int number = stoi(op.substr(2));
            maxpq.push(number);
            maxpq.push(number);
            continue;
        }

        if (op[2] == '-') {
            int max = maxpq.top();
            int min = minpq.top();
            minpq.pop();
            if (max == min) {
                maxpq.pop();
            }
            continue;
        }

        int max = maxpq.top();
        int min = minpq.top();
        maxpq.pop();
        if (max == min) {
            minpq.pop();
        }
    }
    return answer;
}